cmd/agent/app/agent.go

```diff
@@ -86,8 +89,22 @@ func (a *Agent) HTTPAddr() string {
 
 // Stop forces all agent go routines to exit.
 func (a *Agent) Stop() {
+	// first, close the http server, so that we don't have any more inflight requests
+	timeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	if err := a.httpServer.Shutdown(timeout); err != nil {
+		a.logger.Error("failed to close HTTP server", zap.Error(err))
+	}
+	defer cancel()
+
+	// then, close all processors that are called by the incoming http requests
 	for _, processor := range a.processors {
-		go processor.Stop()
+		processor.Stop()
+	}
+
+	// and the reporter that was being used by the processor
+	if a.reporter != nil {
```

Comment:
is nil check necessary?

Reply:
Not in this iteration. Previously, I got a null pointer because I forgot to set the reporter and thought it would be a good idea to not panic in case the agent's reporter wasn't set. I'm removing it though.